src/LagrangianParticleTracking/update_particle_properties.jl

```diff
@@ -128,10 +128,24 @@ function update_particle_properties!(lagrangian_particles, model, Δt)
     events = []
 
     for (field_name, tracked_field) in pairs(lagrangian_particles.tracked_fields)
-        compute!(tracked_field)
-        particle_property = getproperty(lagrangian_particles.properties, field_name)
+        if isnothing(tracked_field)
+            if field_name in keys(model.tracers)
+                tracked_field = getproperty(model.tracers, field_name)
+            elseif field_name in keys(model.auxiliary_fields)
+                tracked_field = getproperty(model.auxiliary_fields, field_name)
+            elseif field_name in (:u, :v, :w)
+                tracked_field = getproperty(model.velocities, field_name)
+            else
+                throw(ArgumentError("$field_name is not a model field that can be tracked"))
+            end
+        else
+            compute!(tracked_field)
+        end
```

Comment:
I'm not sure I entirely follow this. But would it be easier to do something like:

```julia
tracked_field = fields(model)[field_name]
```

?

Reply:
Oh yeah that works great! That's exactly what I was looking for here.

```suggestion
        if isnothing(tracked_field)
           tracked_field = fields(model)[field_name]
        else
            compute!(tracked_field)
        end
```

Reply:
Why is there a branch for `nothing` `tracked_field`? I don't totally understand what's being supported here.

Reply:
There's this issue with tracked fields where (we don't think) you can specify a tracer as a tracked field without a weird work around because the field isn't defined until after the model is defined, but the model can't be defined until the particles have been defined, and this is a solution that directly references the tracer field in the model instead. I wrote perhaps a better explanation of the problem [here](https://github.com/CliMA/Oceananigans.jl/pull/2662#issue-1304970333).

This isn't a particularly nice solution but I'm not sure how else to implement it since `tracked_fields` is a named tuple, maybe instead of `nothing` I could define an empty function such as `Tracer()` (like grids does with `Perodic()` etc.), then do something like:
```
function compute!(comp::Tracer(), field_name, model)
    comp = fields(model)[field_name]
end
compute!(comp::ComputedField, field_name, model) = compute!(comp)
```
Then in place of the above just have `compute!(tracked_field, field_name, model)`